docs/quickstart.rst

```diff
@@ -23,6 +23,24 @@ Fairlearn is also available on
 For checking out the latest version in our repository check out our
 :ref:`advanced_install`.
 
+.. note::
+
+    The Fairlearn API is not completely stable, so example code in 
+    this documentation may not work with every version of Fairlearn.
+    In particular, it may be necessary to install Fairlearn directly
+    from GitHub via:
+
+    .. code-block:: bash
+
+        pip install git+https://github.com/fairlearn/fairlearn.git
```

Comment:
This seems like a good note, and great to include at the top of the API doc pages.  But for a quickstart, my vote is that we set new users up for success - if they blindly copy the code and run it, the code we show them will work, and that code will show them the value of the library.

Until documentation and release processes are more stable (eg, https://github.com/fairlearn/fairlearn/issues/482, https://github.com/fairlearn/fairlearn/pull/551, https://github.com/fairlearn/fairlearn-proposals/pull/13 and https://github.com/fairlearn/fairlearn-proposals/pull/11), what do you think about making lines 9-14 recommend installing from master, and move the other options with this great note into the existing "Advanced installation instructions" page that the quickstart already links to?

I'm all for the more thorough versioning and release process, and I get that this might not feel like a great long-term solution too.  But this isn't a hypothetical problem - it keeps tripping people up (https://github.com/fairlearn/fairlearn/issues/610, https://github.com/fairlearn/fairlearn/issues/530, https://github.com/fairlearn/fairlearn/pull/532, other folks in gitter).  If the quickstart recommends installing from master, it may also set folks up for success with then navigating through the web documentation and being able to try other parts of the API (eg, https://github.com/fairlearn/fairlearn/issues/448, https://github.com/fairlearn/fairlearn/issues/495, https://github.com/fairlearn/fairlearn/issues/552 and https://github.com/fairlearn/fairlearn/issues/556).

Reply:
The reason it trips up folks is that they install with pip and quickstart shows the development version's quickstart. That should be fixed with versioned docs, right? [To my knowledge @riedgar-ms  was planning to default to the latest stable version in the docs once v0.5.0 is out.]

@riedgar-ms if you check this in, then the quickstart of every following version will have this at the top. Does that make sense?

Isn't what you're describing the case from lines 23-24?

Reply:
I agree that we should probably make the default-from-homepage docs be the latest release (and automate that update). The quickstart would then match the version users pip installed.

I think that it does make sense to have some sort of note at the top of every quickstart, warning people to match versions. We get too many bugs of this.

As for the pip install... what I wrote is slightly different, since it's pip that does the git clone into one of its own directories. However, I hadn't noticed that bit, and will redraft to make what it point at those advanced instructions.